packages/core/src/controllers/verify.ts

```diff
@@ -1,76 +1,130 @@
-    if (this.verifyDisabled) return;
-
-    // ignore on non browser environments
-    if (isReactNative() || !isBrowser()) return;
+  get storeKey(): string {
+    return `verify:public:key`;
+  }
 
-    const verifyUrl = this.getVerifyUrl(params?.verifyUrl);
-    // if init is called again with a different url, remove the iframe and start over
-    if (this.verifyUrl !== verifyUrl) {
-      this.removeIframe();
+  public init = async () => {
+    this.publicKey = await this.store.getItem(this.storeKey);
+    if (this.publicKey && toMiliseconds(this.publicKey?.expiresAt) < Date.now()) {
+      this.logger.debug("verify v2 public key expired");
+      await this.removePublicKey();
     }
-    this.verifyUrl = verifyUrl;
-
-    try {
-      await this.createIframe();
-    } catch (error) {
-      this.logger.info(`Verify iframe failed to load: ${this.verifyUrl}`);
-      this.logger.info(error);
-      // if the iframe fails to load, disable verify
-      this.verifyDisabled = true;
+    if (!this.publicKey) {
+      await this.fetchAndPersistPublicKey();
     }
   };
 
   public register: IVerify["register"] = async (params) => {
-    if (!this.initialized) {
-      this.addToQueue(params.attestationId);
-      await this.init();
-    } else {
-      this.sendPost(params.attestationId);
+    if (!isBrowser()) return;
+    const { id, decryptedId } = params;
+    const url = `${this.verifyUrlV2}/attestation?projectId=${this.projectId}`;
+    let src = "";
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify({ id, decryptedId }),
+      });
+      const { srcdoc } = await response.json();
+      src = srcdoc;
+      this.logger.debug("srcdoc fetched", src);
+    } catch (e) {
+      this.logger.warn(e);
+      return;
     }
+    try {
+      const document = getDocument() as Document;
+      const abortTimeout = this.startAbortTimer(ONE_SECOND * 3);
+      const attestationJwt = await new Promise((resolve) => {
+        const abortListener = () => {
+          window.removeEventListener("message", listener);
+          document.body.removeChild(iframe);
+          throw new Error("attestation aborted");
+        };
+        this.abortController.signal.addEventListener("abort", abortListener, {
+          signal: this.abortController.signal,
+        });
+        const iframe = document.createElement("iframe");
+        iframe.srcdoc = src;
+        iframe.style.display = "none";
+        const listener = (event: MessageEvent) => {
+          if (!event.data) return;
+          const data = JSON.parse(event.data);
+          if (data.type === "verify_attestation") {
+            clearInterval(abortTimeout);
+            document.body.removeChild(iframe);
+            this.abortController.signal.removeEventListener("abort", abortListener);
+            window.removeEventListener("message", listener);
+            resolve(data.attestation === null ? "" : data.attestation);
```

Comment:
which are the possible values of `data.attestation`? 

would `resolve(data.attestation || '');` work?

Reply:
I think `resolve(data.attestation ?? '');` would work then